.github/workflows/publish_image.yaml

```
@@ -0,0 +1,72 @@
+name: Publish Image
+
+env:
+  GCLOUD_PROJECT_NUMBER: "755795606445"
+  GCLOUD_PROJECT_NAME: "trading-prod-228918"
+  ARTIFACT_REGISTRY: "europe-west2-docker.pkg.dev"
+  DOCKER_REPOSITORY: "trading-docker-repo"
+
+on:
+  workflow_call:
+    inputs:
+      image_name:
+        required: true
+        type: string
+      build_command:
+        required: true
+        type: string
+
+jobs:
+  action:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - name: SET UP - Checkout Code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: SET UP - Get Commit Tag
+        run: |
+          #!/bin/sh
+          set -ox
```

The `-o` flag for `set` is for setting a nominated option, but there's no such option specified.
Should this look a little different, like so?
```suggestion
          set -xo pipefail
```